private_gpt/components/vector_store/vector_store_component.py

```diff
@@ -135,6 +135,7 @@ def get_retriever(
         similarity_top_k: int = 2,
     ) -> VectorIndexRetriever:
         # This way we support qdrant (using doc_ids) and the rest (using filters)
+        similarity_top_k = self.settings.rag.similarity_top_k
```

Comment:
You are effectively making the similarity_top_k function param useless, because you are overriding it. I'd suggest to respect it if set. Something like this should work:

```
def get_retriever(
        self,
        index: VectorStoreIndex,
        context_filter: ContextFilter | None = None,
        similarity_top_k: int | None = None,
    ) -> VectorIndexRetriever:
        # This way we support qdrant (using doc_ids) and the rest (using filters)
        similarity_top_k = similarity_top_k or self.settings.rag.similarity_top_k
```

There is an alternative which I like more because is more scalable as the project evolves than the current solution of impacting the "get_retriever" for the whole application:

- change `settings.rag.similarity_top_k` to `settings.context_chat_rag.similarity_top_k`
- in chat_service.py, when `get_retriever` is called, pass the `similarity_top_k` param getting it from that setting

That way the setting only applies to the contextual chat use case, and whenever we add more use cases that require a different amount of top_k retrieved, we can fine tune it in different settings

Reply:
I have changed the code, except i left the settings value as "rag" instead of context_chat_rag.  I basically ran out of time, and need to switch gear. But I can change that settings name to context_chat_rag if you prefer that.